### PR TITLE
develop: add Django dev guide

### PR DESCRIPTION
**Pull request by wd0517**
### First-time contributors' checklist 

- [x] I've signed [**Contributor License Agreement**](https://cla-assistant.io/pingcap/docs) that's required for repo owners to accept my contribution.

### Which TiDB version(s) do your changes apply to? (Required)

**Tips for choosing the affected version(s):**

By default, **CHOOSE MASTER ONLY** so your changes will be applied to the next TiDB major or minor releases. If your PR involves a product feature behavior change or a compatibility change, **CHOOSE THE AFFECTED RELEASE BRANCH(ES) AND MASTER**.

For details, see [tips for choosing the affected versions](https://github.com/pingcap/docs/blob/master/CONTRIBUTING.md#guideline-for-choosing-the-affected-versions).

- [x] master (the latest development version)
- [x] v7.4 (TiDB 7.4 versions)
- [x] v7.3 (TiDB 7.3 versions)
- [x] v7.2 (TiDB 7.2 versions)
- [x] v7.1 (TiDB 7.1 versions)
- [x] v7.0 (TiDB 7.0 versions)
- [x] v6.5 (TiDB 6.5 versions)
- [x] v6.1 (TiDB 6.1 versions)
- [ ] v5.4 (TiDB 5.4 versions)
- [ ] v5.3 (TiDB 5.3 versions)
- [ ] v5.2 (TiDB 5.2 versions)
- [ ] v5.1 (TiDB 5.1 versions)
- [ ] v5.0 (TiDB 5.0 versions)

### What is the related PR or file link(s)?

- This PR is translated from:
- Other reference link(s):

### Do your changes match any of the following descriptions?

- [ ] Delete files
- [ ] Change aliases
- [ ] Need modification after applied to another branch 
- [ ] Might cause conflicts after applied to another branch

### CODE REVIEW

**Comment by Icemap** [comment]
This PR will be instead of https://github.com/pingcap/docs/pull/14285.

File: develop/dev-guide-sample-application-python-django.md [hunk 1]
```diff
@@ -0,0 +1,322 @@
...
+    ```shell
+    cp .env.example .env
+    ```
+
+5. Copy and paste the corresponding connection string into the `.env` file. The example result is as follows:
+
+    ```dotenv
+    TIDB_HOST='{host}.clusters.tidb-cloud.com'
+    TIDB_PORT='4000'
+    TIDB_USER='{username}'
```
Comment (sykp241095) [inline develop/dev-guide-sample-application-python-django.md:144]:
```suggestion
    TIDB_USER='{user}'
```

File: develop/dev-guide-sample-application-python-django.md [hunk 2]
```diff
@@ -0,0 +1,322 @@
+---
+title: Connect to TiDB with Django
+summary: Learn how to build a Django application with TiDB. This tutorial gives Python sample code snippets that work with TiDB using Django.
```
Comment (sykp241095) [inline develop/dev-guide-sample-application-python-django.md:3]:
```suggestion
summary: Learn how to build a Django application with TiDB. This tutorial shows Python sample code snippets that work with TiDB using Django.
```

File: develop/dev-guide-sample-application-python-django.md [hunk 3]
```diff
@@ -0,0 +1,322 @@
...
+    ```shell
+    cp .env.example .env
+    ```
+
+6. Copy and paste the corresponding connection string into the `.env` file. The example result is as follows:
+
+    ```dotenv
+    TIDB_HOST='{gateway-region}.aws.tidbcloud.com'
+    TIDB_PORT='4000'
+    TIDB_USER='{prefix}.root'
```
Comment (sykp241095) [inline develop/dev-guide-sample-application-python-django.md:110]:
```suggestion
    TIDB_USER='{user}'
```

**Review by sykp241095 — APPROVED** [review]

File: develop/dev-guide-sample-application-python-django.md [hunk 4]
```diff
@@ -0,0 +1,322 @@
...
+
+In this tutorial, you can learn how to use TiDB and Django to accomplish the following tasks:
+
+- Set up your environment.
+- Connect to your TiDB cluster using Django.
+- Build and run your application. Optionally, you can find sample code snippets for basic CRUD operations.
+
+> **Note:**
+>
+> This tutorial works with TiDB Serverless, TiDB Dedicated, and TiDB Self-Hosted.
```
Comment (winkyao) [inline develop/dev-guide-sample-application-python-django.md:18]:
```suggestion
> This tutorial works with TiDB Serverless, TiDB Dedicated, and TiDB Self-Hosted clusters.
```

File: develop/dev-guide-sample-application-python-django.md [hunk 5]
```diff
@@ -0,0 +1,322 @@
...
+
+## Prerequisites
+
+To complete this tutorial, you need:
+
+- [Python 3.8 or higher](https://www.python.org/downloads/).
+- [Git](https://git-scm.com/downloads).
+- A TiDB cluster.
+
+<CustomContent platform="tidb">
```
Comment (winkyao) [inline develop/dev-guide-sample-application-python-django.md:28]: what are the differences between platform="tidb" and "tidb-cloud"? Seems like the same content.

Comment (wd0517) [reply develop/dev-guide-sample-application-python-django.md:28]: This document will exists in both TiDB Cloud and TiDB, but the following documentation is only available in TiDB, so there are some differences in link address: 
- https://docs.pingcap.com/tidb/stable/quick-start-with-tidb#deploy-a-local-test-cluster
- /quick-start-with-tidb.md#deploy-a-local-test-cluster

File: develop/dev-guide-sample-application-python-django.md [hunk 6]
```diff
@@ -0,0 +1,322 @@
...
+4. Run the following command to copy `.env.example` and rename it to `.env`:
+
+    ```shell
+    cp .env.example .env
+    ```
+
+5. Copy and paste the corresponding connection string into the `.env` file. The example result is as follows:
+
+    ```dotenv
+    TIDB_HOST='{host}.clusters.tidb-cloud.com'
```
Comment (Mini256) [inline develop/dev-guide-sample-application-python-django.md:142]:
```suggestion
    TIDB_HOST='{host}'
```

File: develop/dev-guide-sample-application-python-django.md [hunk 7]
```diff
@@ -0,0 +1,322 @@
...
+
+6. Copy and paste the corresponding connection string into the `.env` file. The example result is as follows:
+
+    ```dotenv
+    TIDB_HOST='{host}'  # e.g. gateway01.ap-northeast-1.prod.aws.tidbcloud.com
+    TIDB_PORT='4000'
+    TIDB_USER='{user}'  # e.g. xxxxxx.root
+    TIDB_PASSWORD='{password}'
+    TIDB_DB_NAME='test'
+    CA_PATH='{ssl_ca}'  # e.g. /etc/ssl/cert.pem (mac)
```
Comment (winkyao) [inline develop/dev-guide-sample-application-python-django.md:113]: Could we use the example of linux instead of mac in the annotation?

**Review by winkyao — COMMENTED** [review]
Rest parts LGTM

**Review by winkyao — APPROVED** [review]
LGTM

**Review by Icemap — APPROVED** [review]

**Review by sykp241095 — APPROVED** [review]

File: develop/dev-guide-sample-application-python-django.md [hunk 8]
```diff
@@ -0,0 +1,323 @@
+---
+title: Connect to TiDB with Django
+summary: Learn how to build a Django application with TiDB. This tutorial shows Python sample code snippets that work with TiDB using Django.
```
Comment (Oreoxmt) [inline develop/dev-guide-sample-application-python-django.md:3]:
```suggestion
summary: Learn how to connect to TiDB using Django. This tutorial gives Python sample code snippets that work with TiDB using Django.
```

File: develop/dev-guide-sample-application-python-django.md [hunk 9]
```diff
@@ -0,0 +1,323 @@
...
+---
+
+# Build a Django App with TiDB
+
+TiDB is a MySQL-compatible database, and [Django](https://www.djangoproject.com/) is a popular web framework for Python, including a powerful ORM library.
```
Comment (Oreoxmt) [inline develop/dev-guide-sample-application-python-django.md:8]:
```suggestion
# Connect to TiDB with Django

TiDB is a MySQL-compatible database, and [Django](https://www.djangoproject.com) is a popular web framework for Python, which includes a powerful Object Relational Mapper (ORM) library.
```

File: develop/dev-guide-sample-application-python-django.md [hunk 10]
```diff
@@ -0,0 +1,323 @@
...
+Run the following commands in your terminal window to clone the sample code repository:
+
+```shell
+git clone https://github.com/tidb-samples/tidb-python-django-quickstart.git
+cd tidb-python-django-quickstart
+```
+
+### Step 2: Install dependencies
+
+Run the following command to install the required packages (including Django, django-tidb and mysqlclient) for the sample app:
```
Comment (Oreoxmt) [inline develop/dev-guide-sample-application-python-django.md:60]:
```suggestion
Run the following command to install the required packages (including Django, django-tidb, and mysqlclient) for the sample app:
```

File: develop/dev-guide-sample-application-python-django.md [hunk 11]
```diff
@@ -0,0 +1,323 @@
...
+
+```shell
+pip install -r requirements.txt
+```
+
+If you encounter installation issues with mysqlclient, refer to the [mysqlclient official documentation](https://github.com/PyMySQL/mysqlclient#install).
+
+#### What is `django-tidb`?
+
+django-tidb is a TiDB dialect for Django that addresses compatibility issues between TiDB and Django, to install it, you need to select the version that corresponds with your Django version, for example, if you are using `django==4.2.*`, you need to install `django-tidb==4.2.*`, the minor release number do not need to be the same, use the latest minor release of each, for more information, refer to [django-tidb repository](https://github.com/pingcap/django-tidb).
```
Comment (Oreoxmt) [inline develop/dev-guide-sample-application-python-django.md:70]:
```suggestion
`django-tidb` is a TiDB dialect for Django that resolves compatibility issues between TiDB and Django.

To install `django-tidb`, choose a version that matches your Django version. For example, if you are using `django==4.2.*`, install `django-tidb==4.2.*`. The minor version does not need to be the same. It is recommended to use the latest minor version.

For more information, refer to [django-tidb repository](https://github.com/pingcap/django-tidb).
```

File: develop/dev-guide-sample-application-python-django.md [hunk 12]
```diff
@@ -0,0 +1,323 @@
...
+    Be sure to replace the placeholders `{}` with the connection parameters, and remove the `CA_PATH` line. If you are running TiDB locally, the default host address is `127.0.0.1`, and the password is empty.
+
+3. Save the `.env` file.
+
+</div>
+</SimpleTab>
+
+### Step 4: Initialize the database
+
+Navigate to the project root directory and run the following command to initialize the database:
```
Comment (Oreoxmt) [inline develop/dev-guide-sample-application-python-django.md:182]:
```suggestion
In the root directory of the project, run the following command to initialize the database:
```

File: develop/dev-guide-sample-application-python-django.md [hunk 13]
```diff
@@ -0,0 +1,323 @@
...
+
+```shell
+python manage.py migrate
+```
+
+### Step 5: Run the sample application
+
+1. Start the app in the development mode:
```
Comment (Oreoxmt) [inline develop/dev-guide-sample-application-python-django.md:190]:
```suggestion
1. Run the application in the development mode:
```

File: develop/dev-guide-sample-application-python-django.md [hunk 14]
```diff
@@ -0,0 +1,323 @@
...
+
+    ```shell
+    python manage.py runserver
+    ```
+
+    The default port is 8000. If you want to change the port, you can add the port number after the command, for example: `python manage.py runserver 8080`
```
Comment (Oreoxmt) [inline develop/dev-guide-sample-application-python-django.md:196]:
```suggestion
    The application runs on port `8000` by default. To use a different port, you can append the port number to the command. The following is an example:

    ```shell
    python manage.py runserver 8080
    ```
```

File: develop/dev-guide-sample-application-python-django.md [hunk 15]
```diff
@@ -0,0 +1,323 @@
...
+
+2. Open your browser and go to `http://localhost:8000/` to see the sample app in action, within the app, you can:
```
Comment (Oreoxmt) [inline develop/dev-guide-sample-application-python-django.md:198]:
```suggestion
2. To access the application, open your browser and go to `http://localhost:8000/`. In the sample application, you can:
```

File: develop/dev-guide-sample-application-python-django.md [hunk 16]
```diff
@@ -0,0 +1,323 @@
...
+    - View all players.
+    - Update a player.
+    - Delete a player.
+    - Trade goods between two players.
+
+## Sample code snippets
+
+You can refer to the following sample code snippets to complete your own application development.
+
+For complete sample code and how to run it, check out the [tidb-samples/tidb-python-django-quickstart](https://github.com/tidb-samples/tidb-python-django-quickstart.git) repository.
```
Comment (Oreoxmt) [inline develop/dev-guide-sample-application-python-django.md:211]:
```suggestion
For complete sample code and how to run it, check out the [tidb-samples/tidb-python-django-quickstart](https://github.com/tidb-samples/tidb-python-django-quickstart) repository.
```

File: develop/dev-guide-sample-application-python-django.md [hunk 17]
```diff
@@ -0,0 +1,323 @@
...
+
+### Configure the database connection
+
+In the file `sample_project/settings.py` add the following code:
```
Comment (Oreoxmt) [inline develop/dev-guide-sample-application-python-django.md:215]:
```suggestion
### Connect to TiDB

In the file `sample_project/settings.py`, add the following configurations:
```

File: develop/dev-guide-sample-application-python-django.md [hunk 18]
```diff
@@ -0,0 +1,323 @@
...
+
+TIDB_CA_PATH = ${ca_path}
+if TIDB_CA_PATH:
+    DATABASES["default"]["OPTIONS"]["ssl_mode"] = "VERIFY_IDENTITY"
+    DATABASES["default"]["OPTIONS"]["ssl"] = {
+        "ca": TIDB_CA_PATH,
+    }
+```
+
+You need to replace `${tidb_host}`, `${tidb_port}`, `${tidb_user}`, `${tidb_password}`, `${tidb_db_name}` and `${ca_path}` with the actual values of your TiDB cluster.
```
Comment (Oreoxmt) [inline develop/dev-guide-sample-application-python-django.md:240]:
```suggestion
You need to replace `${tidb_host}`, `${tidb_port}`, `${tidb_user}`, `${tidb_password}`, `${tidb_db_name}`, and `${ca_path}` with the actual values of your TiDB cluster.
```

File: develop/dev-guide-sample-application-python-django.md [hunk 19]
```diff
@@ -0,0 +1,323 @@
...
+
+class Player(models.Model):
+    name = models.CharField(max_length=32, blank=False, null=False)
+    coins = models.IntegerField(default=100)
+    goods = models.IntegerField(default=1)
+    created_at = models.DateTimeField(auto_now_add=True)
+    updated_at = models.DateTimeField(auto_now=True)
+```
+
+For more information, refer to [Django Models](https://docs.djangoproject.com/en/dev/topics/db/models/).
```
Comment (Oreoxmt) [inline develop/dev-guide-sample-application-python-django.md:255]:
```suggestion
For more information, refer to [Django models](https://docs.djangoproject.com/en/dev/topics/db/models/).
```

File: develop/dev-guide-sample-application-python-django.md
Comment (Oreoxmt) [inline develop/dev-guide-sample-application-python-django.md:1]: Please delete the `develop/dev-guide-outdated-for-django.md` file and add aliases in this file:

```
aliases: ['/tidb/dev/dev-guide-outdated-for-django']
```

File: develop/dev-guide-sample-application-python-django.md [hunk 20]
```diff
@@ -0,0 +1,332 @@
...
+
+# delete multiple objects
+Player.objects.filter(coins=100).delete()
+```
+
+For more information, refer to [Delete data](/develop/dev-guide-delete-data.md).
+
+## Next steps
+
+- Learn more usage of Django from [the documentation of django](https://www.djangoproject.com/).
```
Comment (Oreoxmt) [inline develop/dev-guide-sample-application-python-django.md:326]:
```suggestion
- Learn more usage of Django from [the documentation of Django](https://www.djangoproject.com/).
```

**Review by Oreoxmt — APPROVED** [review]
Rest LGTM

**Pull request merged**